transport/endpoint.go

```diff
@@ -0,0 +1,12 @@
+package transport
+
+import "github.com/aws/smithy-go"
+
+// Endpoint is a Smithy endpoint.
+type Endpoint struct {
+	URI string
+
+	Fields *FieldSet
```

Comment:
**Question**: Why is `Fields` as a type necessary? It seems to be representing headers, why should we not just use [net/http#Header](https://pkg.go.dev/net/http#Header)?